src/fswAlgorithms/opticalNavigation/flybyODuKF/flybyODuKF.cpp

```diff
@@ -0,0 +1,485 @@
+        pXY += kMat;
+    }
+
+    /*! - Then we need to invert the SyT*Sy matrix to get the Kalman gain factor.  Since
+     The Sy matrix is lower triangular, we can do a back-sub inversion instead of
+     a full matrix inversion. Equation 27 in the reference document.*/
+    STkMatT = FlybyODuKF::forwardSubstitution(sy, pXY.transpose());
+    kMat = FlybyODuKF::backSubstitution(sy.transpose(), STkMatT).transpose();
+
+    /*! - Difference the yBar and the observations to get the observed error and
+     multiply by the Kalman Gain to get the state update.  Add the state update
+     to the state to get the updated state value (equation 27).*/
+    this->state = this->xBar + kMat*(this->obs - yBar);
+
+    /*! - Compute the updated matrix U from equation 28 */
+    Eigen::MatrixXd Umat;
+    Umat.setZero(this->state.size(), this->obs.size());
+    Umat = kMat * sy;
+
+    /*! - For each column in the update matrix, perform a cholesky down-date on it to
+     get the total shifted S matrix (called sBar in internal parameters*/
+    for(size_t i=0; i < Umat.cols(); i++)
+    {
+        this->sBar = FlybyODuKF::choleskyUpDownDate(this->sBar, Umat.col(i), -1);
+    }
+
+    /*! - Compute equivalent covariance based on updated sBar matrix*/
+    this->covar = this->sBar*this->sBar.transpose();
+}
+
+/*! This method integrates the equations of motion of two body point mass gravity using Runge-Kutta 4 (RK4)
+    @param interval integration interval
+    @param X0 initial state
+    @param dt time step
+    @return Eigen::VectorXd
+*/
+Eigen::VectorXd FlybyODuKF::propagator(std::array<double, 2> interval, const Eigen::VectorXd& X0, double dt) const
+{
+    double t_0 = interval[0];
+    double t_f = interval[1];
+    double t = t_0;
+    Eigen::VectorXd X = X0;
+
+    std::function<Eigen::VectorXd(double, Eigen::VectorXd)> f = [this](double t, Eigen::VectorXd state)
+    {
+        Eigen::VectorXd stateDerivative(state.size());
+        /*! Implement point mass gravity for the propagation */
+        stateDerivative.segment(0,3) = state.segment(3, 3);
+        stateDerivative.segment(3,3) = - this->muCentral/(pow(state.head(3).norm(),3)) * state.head(3);
+
+        return stateDerivative;
+    };
+
+    /*! Propagate to t_final with an RK4 integrator */
+    double N = ceil((t_f-t_0)/dt);
+    for (int c=0; c < N; c++) {
+        double step = std::min(dt,t_f-t);
+        X = this->rk4(f, X, t, step);
+        t = t + step;
+    }
+
+    return X;
+}
+
+/*! This method integrates the equations of motion of two body point mass gravity using Runge-Kutta 4 (RK4)
+    @param ODEfunction function handle that includes the equations of motion
+    @param X0 initial state
+    @param t0 initial time
+    @param dt time step
+    @return Eigen::VectorXd
+*/
+Eigen::VectorXd FlybyODuKF::rk4(const std::function<Eigen::VectorXd(double, Eigen::VectorXd)>& ODEfunction, const Eigen::VectorXd& X0, double t0, double dt) const
```

Comment:
:large_blue_circle: You could drop each parameter onto a new line like
```
Eigen::VectorXd FlybyODuKF::RK4(const std::function<Eigen::VectorXd(double, Eigen::VectorXd)>& ODEfunction,
                                const Eigen::VectorXd& X0, 
                                double t0, 
                                double dt) const
```

Reply:
Done